cmd/ctr/commands/run/run_unix.go

```diff
@@ -205,10 +205,30 @@ func NewContainer(ctx gocontext.Context, client *containerd.Client, context *cli
 		}
 	}
 
+	runtimeOpts, err := getRuntimeOptions(context)
+	if err != nil {
+		return nil, err
+	}
+	cOpts = append(cOpts, containerd.WithRuntime(context.String("runtime"), runtimeOpts))
+
+	opts = append(opts, oci.WithAnnotations(commands.LabelArgs(context.StringSlice("label"))))
+	var s specs.Spec
+	spec = containerd.WithSpec(&s, opts...)
+
+	cOpts = append(cOpts, spec)
+
+	// oci.WithImageConfig (WithUsername, WithUserID) depends on access to rootfs for resolving via
+	// the /etc/{passwd,group} files. So cOpts needs to have precedence over opts.
+	return client.NewContainer(ctx, id, cOpts...)
+}
+
+func getRuntimeOptions(context *cli.Context) (interface{}, error) {
```

Comment:
This should be `func getRuncOptions(*cli.Context)(&options.Options, error)` and should be called only when `context.String("runtime")` is runc.

Reply:
I was hoping to handle all possible runtime specific options in the same function, although right now there are only runc options, -- that's why it is named `getRuntimeOptions` instead of `getRuncOptions`. Hopefully when we add other runtime specific command line options, we handle it here as well. WDYT?

Reply:
Or maybe I should rewrite it following your suggestions so that
`getRuntimeOptions` calls `getRuncOptions`?

Reply:
@AkihiroSuda please take another look. Thanks!